zhaquirks/tuya/__init__.py

```diff
@@ -54,37 +72,40 @@ class TuyaManufClusterAttributes(TuyaManufCluster):
 
     def handle_cluster_request(self, tsn: int, command_id: int, args: Tuple) -> None:
         """Handle cluster request."""
-        if command_id not in (0x0001, 0x0002):
+        if not (
+            command_id in (0x0001, 0x0002)
+            and len(args) == 1
+            and isinstance(args[0], TuyaManufCluster.Command)
+        ):
             return super().handle_cluster_request(tsn, command_id, args)
 
         tuya_cmd = args[0].command_id
-        tuya_value = args[0].data[1:]  # first uint8_t is length
+        tuya_data = args[0].data
 
         _LOGGER.debug(
             "[0x%04x:%s:0x%04x] Received value %s "
             "for attribute 0x%04x (command 0x%04x)",
             self.endpoint.device.nwk,
             self.endpoint.endpoint_id,
             self.cluster_id,
-            repr(tuya_value),
+            repr(tuya_data[1:]),
             tuya_cmd,
             command_id,
         )
 
         if tuya_cmd not in self.attributes:
             return
 
-        # tuya data is in big endian whereas ztypes use little endian
         ztype = self.attributes[tuya_cmd][1]
-        zvalue, _ = ztype.deserialize(bytes(reversed(tuya_value)))
+        zvalue = tuya_data.to_value(ztype)
         self._update_attribute(tuya_cmd, zvalue)
 
-    def read_attributes(
+    async def read_attributes(
```

Comment:
you can remove `async` if you remove `await`. No point of an extra await

Reply:
pylint was complaining about a redefinition of a derived awaitable function that was no longer awaitable. I can remove it too.

Reply:
ah, yeah, pylint may complain about it. I think we have pylint in tox but not in other checks. It's fine as long as GHA pass

Reply:
So I remove it altogether ?